frame/evm/src/resource.rs

```diff
@@ -0,0 +1,627 @@
+use crate::{AccountCodes, AccountCodesMetadata, Config, Pallet};
+
+use core::marker::PhantomData;
+use evm::{
+	gasometer::{GasCost, StorageTarget},
+	ExitError, Opcode,
+};
+use fp_evm::WeightInfo;
+use sp_core::{Get, H160, H256, U256};
+use sp_runtime::{traits::CheckedAdd, Saturating};
+use sp_std::{collections::btree_map::BTreeMap, vec::Vec};
+
+/// `System::Account` 16(hash) + 20 (key) + 60 (AccountInfo::max_encoded_len)
+pub const ACCOUNT_BASIC_PROOF_SIZE: u64 = 96;
+/// `AccountCodesMetadata` read, temptatively 16 (hash) + 20 (key) + 40 (CodeMetadata).
+pub const ACCOUNT_CODES_METADATA_PROOF_SIZE: u64 = 76;
+/// 16 (hash1) + 20 (key1) + 16 (hash2) + 32 (key2) + 32 (value)
+pub const ACCOUNT_STORAGE_PROOF_SIZE: u64 = 116;
+/// Fixed trie 32 byte hash.
+pub const WRITE_PROOF_SIZE: u64 = 32;
+/// Account basic proof size + 5 bytes max of `decode_len` call.
+pub const IS_EMPTY_CHECK_PROOF_SIZE: u64 = 93;
+/// Storage: Set a new value. (32 bytes)
```

Comment:
Can you elaborate on what this is for? Is it not just a special case of `WRITE_PROOF_SIZE`?

Reply:
Please Check new version